### ISSUE TITLE
Bug: [no-unused-vars] <should consider use in JSDoc>

### ISSUE THREAD

**Issue by parloti** [typescript-eslint/typescript-eslint#8258]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=JYWwDg9gTgLgBAbzgeQHYBECmBnGUICecAvnAGb4hwDkAAgIaoDmArgDb1QD0AxtJtQDcAWABQY4KhiYoZej0xwAkkqkzEYscU2iuAKj1i4exLTaSA1igw48hEqfOorKtVG2jjXMZgAekWHIWVB4YYAhUOAAxYIAKAEoNTzgoTBgWKEiABhFRDyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXWZRAE0IDcBDaVDSfAMxcgBpxtIABAFwCeABxQBjaAEth-QiniTc-APQFipclToN0URNGj5okbgF8QpoA&tsconfig=&tokens=false

### Repro Code

```TypeScript
import { OnDestroy } from '@angular/core';

interface IInter {

}

/**
 * {@link OnDestroy } {@link IInter}
 */
export function Fun() {
  return 0;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected none of these errors to occur:

```
'OnDestroy' is defined but never used. 1:10 - 1:19
'IInter' is defined but never used. 3:11 - 3:17
```

### Actual Result

The linter throws the following errors:

```
'OnDestroy' is defined but never used. 1:10 - 1:19
'IInter' is defined but never used. 3:11 - 3:17
```

**Comment by Josh-Cena** [comment]
typescript-eslint doesn't handle JSDoc. See https://github.com/gajus/eslint-plugin-jsdoc/issues/858 and try `jsdoc/no-undefined-types` instead.
